addon/components/cfb-form-list.js

```diff
@@ -1,29 +1,55 @@
+import { action } from "@ember/object";
 import Component from "@glimmer/component";
 import { tracked } from "@glimmer/tracking";
 import { queryManager } from "ember-apollo-client";
+import { timeout } from "ember-concurrency";
 import { restartableTask } from "ember-concurrency-decorators";
 
-import formListQuery from "ember-caluma/gql/queries/form-list";
+import calumaQuery from "ember-caluma/caluma-query";
+import { allForms } from "ember-caluma/caluma-query/queries";
 
 export default class ComponentsCfbFormListComponent extends Component {
   @queryManager apollo;
+  @calumaQuery({ query: allForms, options: { pageSize: 20 } }) formsQuery;
 
   @tracked _filter = "active";
+  @tracked search = "";
 
   get filter() {
     return this._filter;
   }
   set filter(value) {
     this._filter = value;
-    this.forms.perform();
+    this.fetchForms.perform();
+  }
+
+  @action
+  searchForms(event) {
+    // search input
+    this.search = event.target.value;
+    this.fetchForms.perform();
+  }
+
+  @action
+  loadMoreForms() {
+    this.fetchForms.perform(true);
   }
 
   @restartableTask
-  *forms() {
-    if (this.args.forms) {
-      return this.args.forms.perform();
+  *fetchForms(loadMore = false) {
```

Comment:
I don't really see why this task is needed anymore. This should all be handled by the query object. Let's discuss this!